src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringAfterJoin = '';
+    const finallySeparator = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
```

Comment:
Do we really need this?
If length === 0 we will skip the loop and return `stringAfterJoin` which is equal to ''